transactionoutbox-core/src/main/java/com/gruelbox/transactionoutbox/TransactionOutboxImpl.java

```diff
@@ -416,4 +430,66 @@ public <T> T schedule(Class<T> clazz) {
       return TransactionOutboxImpl.this.schedule(clazz, uniqueRequestId);
     }
   }
+
+  @Override
+  public int unblockAll() {
+    if (!initialized.get()) {
+      throw new IllegalStateException("Not initialized");
+    }
+    if (!(transactionManager instanceof ThreadLocalContextTransactionManager)) {
+      throw new UnsupportedOperationException(
+          "This method requires a ThreadLocalContextTransactionManager");
+    }
+    log.info("Unblocking entries for retry.");
+    try {
+      return ((ThreadLocalContextTransactionManager) transactionManager)
+          .requireTransactionReturns(tx -> persistor.unblockAll(tx));
+    } catch (Exception e) {
+      throw (RuntimeException) Utils.uncheckAndThrow(e);
+    }
+  }
+
+  @Override
+  @SuppressWarnings({"unchecked", "rawtypes"})
+  public int unblockAll(Object transactionContext) {
+    if (!initialized.get()) {
+      throw new IllegalStateException("Not initialized");
+    }
+    if (!(transactionManager instanceof ParameterContextTransactionManager)) {
+      throw new UnsupportedOperationException(
+          "This method requires a ParameterContextTransactionManager");
+    }
+    log.info("Unblocking entries for retry");
+    try {
+      if (transactionContext instanceof Transaction) {
+        return persistor.unblockAll((Transaction) transactionContext);
+      }
+      Transaction transaction =
+          ((ParameterContextTransactionManager) transactionManager)
+              .transactionFromContext(transactionContext);
+      return persistor.unblockAll(transaction);
+    } catch (Exception e) {
+      throw (RuntimeException) Utils.uncheckAndThrow(e);
+    }
+  }
+
+  @Override
+  public List<TransactionOutboxEntry> getBlockedEntries(int page, int batchSize) {
```

Comment:
We already have the `processNow` method, so in order to make this flexible, I think this method should only return the records (without locking or processing them), and then if the caller wants to reprocess them, they can do so explicitly.